### PR TITLE
Stop gracefully wpad while sysupgrade(ing)

### PR DESCRIPTION
**Pull request by tiagofreire-pt**
As suggested by @dimfishr on https://github.com/dimfishr/openwrt/commit/8fc221b4c852bc4fd7cc336f44d02bbc3ffc4432#diff-429fb11e5227b788af214dec9bab84f0e67920ec006ace4c8b4e056a4ae5a04b, this seems to solve the random issue of failing ```sysupgrade``` due to the time ```ẁpad``` takes to stop with ```ath11```drivers.

### CODE REVIEW

**Comment by Ansuel** [comment]
IMHO we don't need this... @robimarko imho we should just apply my hack patch on ath11k and be done with it while we find more time to investigate why ath11k is behaving like that...

**Comment by tiagofreire-pt** [comment]
This is just a quality-of-life patch, due to users claiming the sysupgrade does not work consistently on many devices.

I'm in favor of a root-cause intended solution, instead. Until then, I assume this simple patch circumvents the issue.

File: target/linux/ipq807x/base-files/lib/upgrade/platform.sh [hunk 1]
```diff
@@ -36,6 +36,13 @@ platform_do_upgrade() {
 	redmi,ax6|\
 	xiaomi,ax3600|\
 	xiaomi,ax9000)
+	
+		# Circunvent a random issue with WPAD taking a huge time to stop while upgrading with ath11 driver, leading the sysupgrade role to failure
```
Comment (robimarko) [inline target/linux/ipq807x/base-files/lib/upgrade/platform.sh:40]: Break this into multiple lines

File: target/linux/ipq807x/base-files/lib/upgrade/platform.sh [hunk 2]
```diff
@@ -36,6 +36,13 @@ platform_do_upgrade() {
...
+		/etc/init.d/wpad stop 2>/dev/null
```
Comment (robimarko) [inline target/linux/ipq807x/base-files/lib/upgrade/platform.sh:41]: service

**Comment by robimarko** [comment]
@Ansuel The thing is that upstream had 0 comments on your RFC, and I seriously don't want to get in a situation where OpenWrt depends on a patch that is not upstream.
I am not overly happy with the situation, but personally, I find it much easier to have a hack similar to this(As this is only targeting few boards and in a weird way) for now

**Comment by Ansuel** [comment]
Thing is that this use wpad stop and some arbitrary sleep may still not work if the thing got stuck enough time. But I agree that both solutions are bad.

File: target/linux/ipq807x/base-files/lib/upgrade/platform.sh (hunk 2 above)
Comment (tiagofreire-pt) [reply target/linux/ipq807x/base-files/lib/upgrade/platform.sh:42]: I assume is to force any zombie wpad thread to terminate, as a redundancy to the previous stop command.

**Comment by rmandrad** [comment]
it is a workaround ... but saying this i thought this was not a problem anymore ... i don't get on the dynalink-dlwrx36 nor the qnap ... maybe because there are not enough connections ... worth imho asking in the ax3600 forum ?

**Comment by McGiverGim** [comment]
I can confirm that the problem still persist in the latest versions with my AX3600. I don't know if this patch will work.

**Comment by tiagofreire-pt** [comment]
@Ansuel , were you referring this alternative "patch"? https://github.com/bitthief/openwrt/commit/34735c1cbaf4ebf3818b286440c576961d45a903

**Comment by kirdesde** [comment]
A target specific workaround in the plaform.sh won't work in my opinion, as sysupgrade reboots during stage2 if wpad can't be stopped/killed.
Target specific platform.sh isn't even called in this case. 
see https://github.com/robimarko/openwrt/issues/55#issuecomment-1227479089

**Comment by robimarko** [comment]
@kirdesde Thanks for reminding of that, yeah its all happening way too late for per target stuff

**Pull request closed without merging**